shell.nix

```diff
@@ -0,0 +1,34 @@
+{pkgs ? import <nixpkgs> {}, ...}:
+pkgs.mkShell {
+  buildInputs = with pkgs.buildPackages; [
+    pkg-config
+    openssl
+
+    (rust-bin.fromRustupToolchainFile ./rust-toolchain.toml)
```

Comment:
could it use the toolchain file from the root? like `../rust-toolchain.toml`? then we wouldn't need the toolchain in two spots.

Or is it possible to copy the file in for the build or something like that?